src/views/DogEdit/DogEdit.js

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from 'react';
+import { fetchDogById, updateDog } from '../../services/dogs';
+import { useHistory } from 'react-router-dom';
+import DogForm from '../../components/DogForm/DogForm';
+import './DogEdit.css';
+
+export default function DogEdit(props) {
+  const id = props.match.params.id;
+  const [name, setName] = useState('');
+  const [breed, setBreed] = useState('');
+  const [image, setImage] = useState('');
+  const [bio, setBio] = useState('');
+  const [message, setMessage] = useState('');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await fetchDogById(id);
+      setName(data.name);
+      setBreed(data.breed);
+      setImage(data.image);
+      setBio(data.bio);
+    };
+    fetchData();
+  }, [id]);
+
+  const history = useHistory();
+
+  const handleSubmit = async (e) => {
+    try {
+      e.preventDefault();
+      await updateDog(id, name, breed, image, bio);
+      setMessage('Success! Redirecting...');
+      setTimeout(() => {
+        history.goBack();
+      }, 2500);
+    } catch {
+      setMessage('Oh no! Something went wrong!');
+    }
+  };
+
+  return (
+    <div>
+      <p className="message">{message}</p>
```

Comment:
If you set the initial message state to `null `and use `{message && message}` you can avoid having an empty `<p>` on the page

Reply:
added!